src/pybind/mgr/dashboard_v2/settings.py

```diff
@@ -0,0 +1,113 @@
+# -*- coding: utf-8 -*-
+from __future__ import absolute_import
+
+import errno
+import inspect
+from six import with_metaclass
+
+from . import mgr
+
+
+class Options(object):
+    """
+    If you need to store some configuration value please add the config option
+    name as a class attribute to this class.
+    """
+    # Grafana settings
+    GRAFANA_API_HOST = ('localhost', str)
+    GRAFANA_API_PORT = (3000, int)
+    GRAFANA_API_USERNAME = ('admin', str)
+    GRAFANA_API_PASSWORD = ('admin', str)
+    GRAFANA_API_SCHEME = ('http', str)
+
+    # RGW settings
+    RGW_API_HOST = ('', str)
+    RGW_API_PORT = (80, int)
+    RGW_API_ACCESS_KEY = ('', str)
+    RGW_API_SECRET_KEY = ('', str)
+    RGW_API_ADMIN_RESOURCE = ('admin', str)
+    RGW_API_USER_ID = ('', str)
+    RGW_API_SCHEME = ('http', str)
+
+
+class SettingsMeta(type):
+
+    def __getattribute__(cls, attr):
+        if not attr.startswith('_') and hasattr(Options, attr):
+            default, stype = getattr(Options, attr)
+            return stype(mgr.get_config(attr, default))
+        return SettingsMeta.__dict__.get(attr, None)
+
+    def __setattr__(cls, attr, value):
+        if not attr.startswith('_') and hasattr(Options, attr):
+            mgr.set_config(attr, str(value))
+        else:
+            setattr(SettingsMeta, attr, value)
+
+
+# pylint: disable=no-init
+class Settings(with_metaclass(SettingsMeta, object)):
+    pass
```

Comment:
@rjfd https://github.com/ceph/ceph-ci/commit/082c0b12eb75a6fe3b2b4f783fb4aa7c00507857

Reply:
I had a similar issue and solved it by using [`@add_metaclass`](https://pythonhosted.org/six/#six.add_metaclass)

Reply:
What I meant was: I got that exception, too. I fixed that locally by using `@add_metaclass` instead.